### PR TITLE
Removing Bitly credentials from JS, setting them using application config

### PR DESCRIPTION
**Pull request by xavijam**
No more credentials set within url_shortener class, they will be set from the cdb.config model and those values will be provided through the frontend_config from the backend.

@luisbosque @javisantana

### CODE REVIEW

**Comment by xavijam** [comment]
Sounds good?

File: lib/assets/javascripts/cartodb/common/url_shortener.js [hunk 1]
```diff
@@ -1,29 +1,33 @@
...
   var cachedUrl = this.localStorage.search(originalUrl);
+  var self = this;
+
   if (cachedUrl) {
     return callbacks.success(cachedUrl);
   }
 
-  var self = this;
+  if (!this._LOGIN || !this._KEY) {
+    return callbacks.error(originalUrl);
```
Comment (javisantana) [inline lib/assets/javascripts/cartodb/common/url_shortener.js]: why error?

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -80,6 +80,8 @@ def frontend_config
       oauth_gdrive:               Cartodb.config[:oauth]['gdrive']['client_id'],
       oauth_instagram:            Cartodb.config[:oauth]['instagram']['app_key'],
       oauth_mailchimp:            Cartodb.config[:oauth]['mailchimp']['app_key'],
+      bitly_login:                Cartodb.config[:bitly]['login'],
```
Comment (javisantana) [inline app/helpers/application_helper.rb]: could we make this optional, if we do this in this way everyone will need to add that configuration

Comment (xavijam) [inline app/helpers/application_helper.rb]: You mean:

```rb
...
bitly_login: Cartodb.config[:bitly].present ? Cartodb.config[:bitly]["login"] : "",
bitly_key: Cartodb.config[:bitly].present ? Cartodb.config[:bitly]["key"] : "",
...
```

File: lib/assets/javascripts/cartodb/common/url_shortener.js (hunk 1 above)
Comment (xavijam) [inline lib/assets/javascripts/cartodb/common/url_shortener.js]: Choosing between success or error, it sounds me as an error, but yeah, success could fit as well.

**Comment by luisbosque** [comment]
Yep, bitly should be used only if the login/key exists and it's not empty. Otherwise the real CartoDB map url should be returned instead of the bitly url.

File: app/helpers/application_helper.rb (hunk 2 above)
Comment (javisantana) [inline app/helpers/application_helper.rb]: yes

**Comment by xavijam** [comment]
Until this https://github.com/CartoDB/cartodb-platform/issues/1055 issue is not ready, we can't deploy this pull request.

**Comment by xavijam** [comment]
Apart from that, everything ok @luisbosque @javisantana ?

**Comment by luisbosque** [comment]
:+1:

**Pull request merged**